components/gsma-authenticators/src/main/java/com/wso2telco/gsma/authenticators/voice/VoiceCallAuthenticator.java

```diff
@@ -0,0 +1,265 @@
+                postRequest(onBoardUserUrl, verifyAndRegistartionPostData ,
+                        futureCallback);
+                response.sendRedirect(url);
+            }
+        } catch (IOException e) {
+            log.error("Error occurred while redirecting request", e);
+        } catch(JSONException je){
+            je.printStackTrace();
+        } catch(SQLException sqle){
+
+        } catch(AuthenticatorException ae){
+
+        }
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+        log.info("Tagaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        String queryParams = FrameworkUtils.getQueryStringWithFrameworkContextId(authenticationContext.getQueryParams(),
+                authenticationContext.getCallerSessionKey(), authenticationContext.getContextIdentifier());
+        ApplicationConfig applicationConfig = authenticationContext.getSequenceConfig().getApplicationConfig();
+        Map<String, String> paramMap = Util.createQueryParamMap(queryParams);
+        String msisdn = (String) authenticationContext.getProperty(MSISDN);
+        String clientId = paramMap.get(CLIENT_ID);
+        String applicationName = applicationConfig.getApplicationName();
+        String sessionKey = authenticationContext.getCallerSessionKey();
+
+        boolean isUserExists =false;
+        try {
+            isUserExists= AdminServiceUtil.isUserExists(msisdn);
+        } catch (UserStoreException e) {
+            e.printStackTrace();
+        }
+        if(!isUserExists){
+            String operator = (String) authenticationContext.getProperty(Constants.OPERATOR);
+            try {
+                new UserProfileManager().createUserProfileLoa2(msisdn, operator, Constants.SCOPE_MNV);
+            } catch (UserRegistrationAdminServiceIdentityException e) {
+                e.printStackTrace();
+            } catch (RemoteException e) {
+                e.printStackTrace();
+            }
+        }
+
+
+
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest httpServletRequest) {
+        log.info("Jabaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        return true;
+    }
+
+    @Override
+    public String getContextIdentifier(HttpServletRequest httpServletRequest) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return Constants.VOICECALL_AUTHENTICATOR_NAME;
+    }
+
+    @Override
+    public String getFriendlyName() {
+        return Constants.VOICECALL_AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+
+    private boolean checkUserIsEnrolledValidSoftServer(String url, StringEntity postData) throws IOException {
+        HttpClient httpClient = new DefaultHttpClient();
+        HttpPost httpPost = new HttpPost(url);
+        postData.setContentType("application/json");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "application/json" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS1192)